### ISSUE TITLE
Span.similarity(Token) triggers TypeError

### ISSUE THREAD

**Issue by wizzwizz4** [explosion/spaCy#5356]
## How to reproduce the behaviour
```python
>>> import spacy
>>> nlp = spacy.load("en_core_web_md")
>>> a = nlp("hi")[0]
>>> b = nlp("hello you there")[1:]
>>> a.similarity(b)
0.38044035
>>> b.similarity(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "span.pyx", line 329, in spacy.tokens.span.Span.similarity
TypeError: 'spacy.tokens.token.Token' object is not subscriptable
```

## Info about spaCy

* **spaCy version:** 2.2.4
* **Platform:** Linux-5.2.0-3-amd64-x86_64-with-debian-10.3
* **Python version:** 3.7.3

## Notes
There are 16 different possibilities here; it's possible some of them might've slipped through the test suite. I recommend adding some more sanity checks.

**Comment by adrianeboyd** [comment]
Ah, we'd fixed this on `develop` and just need to backport it. Thanks for the report!

**Comment by svlandeg** [comment]
Will be fixed from 2.3 onwards !
